bindings/python/samples/runtext.py

```diff
@@ -9,23 +9,27 @@ class RunText(SampleBase):
     def __init__(self, *args, **kwargs):
         super(RunText, self).__init__(*args, **kwargs)
         self.parser.add_argument("-t", "--text", help="The text to scroll on the RGB LED panel", default="Hello world!")
+        self.parser.add_argument("-s", "--speed", help="Approximate letters per second.", default="-s0",type=float)
+        #self.parser.add_argument("--led-panel-type", action="store", help="Needed to initialize special panels. Supported: 'FM6126A'", default="", type=str)
 
-    def run(self):
+    def run(self, s):
```

Comment:
If you now pass the speed as paremter to run, you have to call run with this parameter, but this is not expected.

Instead, just read the speed parameter from the flag (as you did before) and calculate the time by doing something like `1.0 / speed`. So if speed is 20, then you get 0.05 (as before), but if speed is 10, you get (0.1 ... longer time wait).

Reply:
Some times I  want the speed to be 0, and sometimes have it run across the panel. If I use time.sleep(1.0/speed),then I can't divide by 0. In my situation, I plan to call this function with different text and different speeds (although the speed will be 0 most of the time).

Reply:
The `run()` function is called from the thread in samplebase.py, so you need to keep it wit no parameters.

If speed is zero, you essentially simply don't want any movement, so in that case, don't update `pos`. In the sleep function you can then do something like `if speed == 0 then 10 else 1.0 / speed`.

(Also important to consider: if you don't have any scrolling, you probably want to move the initial pos to a place where it is visible)